force-app/main/default/classes/ALLO_AllocationsRecalculateService.cls

```diff
@@ -170,6 +170,11 @@ public inherited sharing class ALLO_AllocationsRecalculateService {
             return;
         }
 
+        // Allocations should not be copied to target if source amount is zero and convert to percentage is true
+        if (sourceAmount == 0 && convertToPercentages) {
```

Comment:
This is me being picky, but could we make this an `else if` with the above `if` block? Alternatively, all of these conditions could be in the same `if` but that might look cluttered.